src/components/shared/AutoSave/index.tsx

```diff
@@ -0,0 +1,24 @@
+import debounce from 'lodash/debounce';
+import { useEffect, useCallback, useRef } from 'react';
+
+type AutoSaveProps = {
+  values: any;
+  onSave: () => void;
+};
+
+const AutoSave = ({ values, onSave }: AutoSaveProps) => {
+  const debounceSave = useCallback(debounce(onSave, 1000), [onSave]);
+  const initialValues = useRef(values);
+
+  useEffect(() => {
+    // Don't save until values are changed from initial values (avoids save on initial render)
+    if (initialValues.current === values) {
```

Comment:
This appears to be working as intended. I do feel a little bit concerned because checking for object equality can be tricky.

For example:
```
{ id: '' } === { id: '' }

// --> false
```

 I'm guessing the react hook doesn't actually create a new object, so that's why it's still equal.

Reply:
We'll use lodash isEqual which does deep object equality